[AOTInductor] Add function to free buffer

muchulee8 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #149161
* #149249

Summary:
We add a function that allows users to free the unused buffer.

Test Plan:
Testing correctness:
    python test/inductor/test_aot_inductor.py -k free_inactive

    Testing memory consumption:
    LD_LIBRARY_PATH=/data/users/$USER/pytorch/build/lib
    /home/$USER/local/pytorch/build/bin/test_aoti_inference

Reviewers:

Subscribers:

Tasks:

Tags:

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @chenyang78 @kadeng @amjames @chauhang @aakhundov

muchulee8 committed:
[WIP] [AOTInductor] Add function to free buffer

Summary:
We add a function that allows users to free the unused buffer.

Test Plan:
TBD

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

muchulee8 committed:
Update on "[WIP] [AOTInductor] Add function to free buffer"


Summary:
We add a function that allows users to free the unused buffer.

Test Plan:
TBD

Reviewers:

Subscribers:

Tasks:

Tags:

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng amjames chauhang aakhundov

[ghstack-poisoned]

chenyang78 commented on torch/csrc/inductor/aoti_runner/model_container_runner.cpp:
```diff
@@ -64,6 +64,10 @@ AOTIModelContainerRunner::AOTIModelContainerRunner(
+  free_inactive_constant_buffer_func_ =
+      reinterpret_cast<decltype(free_inactive_constant_buffer_func_)>(
+          model_so_->sym(
+              "AOTInductorModelContainerFreeInactiveConstantBuffer"));
```
What if we load old .so files that were generated before this change?

muchulee8 replied:
Updated to use try.

jbschlosser replied:
@muchulee8 I'm now seeing a somewhat ominous `Could not dlsym AOTInductorModelContainerFreeInactiveConstantBuffer` when loading an old .so; I tracked it down to this PR.

Maybe it makes sense to indicate in the error message that this is okay and the loaded model is still usable?

muchulee8 committed:
Update on "[AOTInductor] Add function to free buffer"


Summary:
We add a function that allows users to free the unused buffer.

Test Plan:
Testing correctness:
    python test/inductor/test_aot_inductor.py -k free_inactive

    Testing memory consumption:
    LD_LIBRARY_PATH=/data/users/$USER/pytorch/build/lib
    /home/$USER/local/pytorch/build/bin/test_aoti_inference

Reviewers:

Subscribers:

Tasks:

Tags:

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng amjames chauhang aakhundov

[ghstack-poisoned]

muchulee8 committed:
Update on "[AOTInductor] Add function to free buffer"


Summary:
We add a function that allows users to free the unused buffer.

Test Plan:
Testing correctness:
    python test/inductor/test_aot_inductor.py -k free_inactive

    Testing memory consumption:
    LD_LIBRARY_PATH=/data/users/$USER/pytorch/build/lib
    /home/$USER/local/pytorch/build/bin/test_aoti_inference

Reviewers:

Subscribers:

Tasks:

Tags:

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng amjames chauhang aakhundov

[ghstack-poisoned]

muchulee8 committed:
Update on "[AOTInductor] Add function to free buffer"


Summary:
We add a function that allows users to free the unused buffer.

Test Plan:
Testing correctness:
    python test/inductor/test_aot_inductor.py -k free_inactive

    Testing memory consumption:
    LD_LIBRARY_PATH=/data/users/$USER/pytorch/build/lib
    /home/$USER/local/pytorch/build/bin/test_aoti_inference

Reviewers:

Subscribers:

Tasks:

Tags:

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng amjames chauhang aakhundov

[ghstack-poisoned]

muchulee8 commented:
@pytorchbot merge

desertfire commented:
Trunk is failing due to a merge conflict with https://github.com/pytorch/pytorch/pull/147105. https://github.com/pytorch/pytorch/pull/149401 is the forward fix.
